src/ChannelServer/Scripting/ScriptManager.cs

```diff
@@ -611,6 +616,39 @@ private int logdebug(IntPtr L)
 			return 0;
 		}
 
+		private int addspawn(IntPtr L)
+		{
+			SpawnData spawnData = new SpawnData();
+
+			spawnData.spawnName = Melua.luaL_checkstring(L, 1);
+			spawnData.mapName = Melua.luaL_checkstring(L, 2);
+			spawnData.monsterName = Melua.luaL_checkstring(L, 3);
+			spawnData.xMin = Melua.luaL_checkinteger(L, 4);
+			spawnData.xMax = Melua.luaL_checkinteger(L, 5);
+			spawnData.zMin = Melua.luaL_checkinteger(L, 6);
+			spawnData.zMax = Melua.luaL_checkinteger(L, 7);
+			spawnData.count = Melua.luaL_checkinteger(L, 8);
+			spawnData.countVariation = Melua.luaL_checkinteger(L, 9);
+			spawnData.respawnTime = Melua.luaL_checkinteger(L, 10);
+
+			spawnData.isFixedLocation = Melua.luaL_checkinteger(L, 11) > 0;
+			var x = (float)Melua.luaL_checknumber(L, 12);
+			var y = (float)Melua.luaL_checknumber(L, 13);
+			var z = (float)Melua.luaL_checknumber(L, 14);
+			spawnData.Position = new Position(x, y, z);
+
+			var direction = Melua.luaL_checkinteger(L, 15);
+			spawnData.Direction = new Direction(direction);
```

Comment:
Does a spawn need a direction? From what I can tell this value isn't used at all.